### PR TITLE
[REFACTOR] Review record ids usage

### PR DESCRIPTION
**Pull request by frascuchon**
This PR renames the custom id (defined in the record model as  `external_id`) to the main record `id`, and exposes the read-only argilla server `id` as the `_server_id` property.

This change normalizes the usage of record ids and provides a unified way to work with them: `id` property.

### CODE REVIEW

**Review by sdiazlor — APPROVED** [review]

**Comment by sdiazlor** [comment]
@burtenshaw Should we remove the _source/tutorials files as they are completely outdated?

**Comment by burtenshaw** [comment]
> @burtenshaw Should we remove the _source/tutorials files as they are completely outdated?

Yes.

File: src/argilla_sdk/records/_resource.py [hunk 1]
```diff
@@ -446,16 +453,25 @@ def to_dict(self) -> Dict[str, List[float]]:
...
+
+    def __len__(self) -> int:
+        return len(self.__metadata_models)
+
+    def __repr__(self) -> str:
+        return self.to_dict().__repr__()
+
     def __init__(self, metadata: Optional[Dict[str, MetadataValue]] = None) -> None:
         self.__metadata_map = metadata or {}
         for key, value in self.__metadata_map.items():
```
Comment (burtenshaw) [inline src/argilla_sdk/records/_resource.py:477]: `__init__` should be first.

**Review by burtenshaw — APPROVED** [review]
This looks good. One high level question: should we be integration testing on `_server_id` ?

**Comment by frascuchon** [comment]
> This looks good. One high level question: should we be integration testing on `_server_id` ?

Yes, totally. I will create one integration test to update records which include only `_server_id`

**Comment by frascuchon** [comment]
> This looks good. One high level question: should we be integration testing on `_server_id` ?

https://github.com/argilla-io/argilla-python/pull/197/files#diff-e1a966e27b39908b708b0e3478007594b860cb161065e91e958b42276d552c1cR114

**Pull request merged**